### PR TITLE
Marina

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
```SUMMARY: Content "Added marina."```

#### Purpose of change
More content for lakes.

#### Describe the solution
Added marina overmap special. Made it rarely spawn on lake shores. There are parking lots, several docks (obviously), a restaurant, small admin building and several tiny park zones.
Added metal bitts furniture and placed it on the marina docks.
Also fixed docks and bridges move cost being too high - reduced `move_cost` from 4 to 2, it's not a hard to cross terrain after all.

#### Additional context
Marina is empty for now - I hesitated to fill it with canoes, kayaks or rafts. We need to add yachts, catamarans, trimarans and other types of serious vessels and made it spawn in marina. That could be done in other PRs.

изображение
изображение
изображение
изображение
изображение

### CODE REVIEW

**Comment by l29ah** [comment]
Shouldn't it spawn on rivers as well?

File: data/json/mapgen/marina.json [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+          [ "t_grass_tall", 5 ],
+          [ "t_grass_long", 3 ],
+          [ "t_dirt", 5 ],
+          [ "t_shrub", 2 ],
+          [ "t_tree", 1 ]
+        ],
+        "^": [ "t_shrub" ],
+        "~": [ "t_floor" ],
+        "&": [ "t_floor" ],
+        "`": [ [ "t_water_dp", 20 ], "t_water_sh" ],
```
Comment (ralreegorganon) [inline data/json/mapgen/marina.json]: You probably want to just make this exclusively deep water. Given how boats currently interact with shallow water (unable to move under their own power unless they also have wheels), I think trying to dock a boat in there would be an exercise in frustration, dodging random shallow water. 

I also just think the slips for the boats would generally be uniformly deep (in terms of our contrast between deep and shallow in the game). If it's done for aesthetic purposes here, we might just want to investigate suggesting some tileset makers make some water variations.

Comment (Night-Pryanik) [reply data/json/mapgen/marina.json]: > If it's done for aesthetic purposes here

Yeah, that's the main reason.
I'll make all water tiles deep ones for now.

**Comment by Night-Pryanik** [comment]
> Shouldn't it spawn on rivers as well?

Yes, it should in theory, but it would require creating almost exact copy of it, the only difference would be river locations in overmap special definitions instead of lake ones. I'm not really interested in doing this right now, maybe later in some subsequent PR.

**Pull request merged**